src/constants/mobile-devices.constants.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @description
+ * This pattern matches any string of three uppercase letters followed by a hyphen.
+ * @example huaweiDevicesRegex.test("ALP-") // returns true
+ */
+export const huaweiDevicesRegex = /\b([A-Z]{3}-)\b/gi;
+
+/**
+ * @regex /\b([A-Z]{3}-)\b/gi
+ * @description
+ * This pattern matches any string that contains a sequence of three uppercase letters followed by a hyphen.
+ * The sequence must be a word on its own (i.e., it must be surrounded by word boundaries).
+ * The 'g' flag is used for global search (to find all matches rather than stopping after the first match), and the 'i' flag is used for case-insensitive search.
+ * @example huaweiDevicesRegex.test("AMN-") // returns true
+ * @example huaweiDevicesRegex.test("ANA-") // returns true
+ * @example huaweiDevicesRegex.test("ANE-") // returns true
+ */
+export const validCodes = new Set([
```

Comment:
Can this be a regex instead of a list of constants? Reason: we do not expect to use them individually as runtime value. And since you are looping to each of them with some operations like upper-string etc - might as well just do a global regex with ignoring case sensitive flag `i`